R/arc-open.R

```diff
@@ -112,3 +149,59 @@ arc_open <- function(url, token = arc_token()) {
 
   res
 }
+
+#' Test if the vector is a content url of the supplied type
+#' @keywords internal
+is_arc_content_url <- function(x, type = c("search", "item", "group", "user")) {
+  pattern <- paste0("arcgis\\.com/home/(", paste0(type, collapse = "|"), ")\\.html")
+  is.character(x) & grepl(pattern, x, perl = TRUE)
+}
+
+#' Convert a URL into a content URL
+#' @keywords internal
+arc_url_item <- function(
+    id,
+    key = "id",
+    base_url = "https://www.arcgis.com/sharing/rest/content",
+    path = "items",
+    call = caller_env()) {
+  if (is_url(id)) {
+    id <- httr2::url_parse(id)[["query"]][[key]]
+  }
+
+  check_string(id, call = call)
+
+  paste(base_url, path, id, sep = "/")
+}
+
+#' Extract the content URL type
+#' @noRd
+arc_content_url_type <- function(url) {
+  str_extract(url, "(?<=arcgis\\.com/home/)[[:lower:]]+(?=\\.html)")
```

Comment:
This regex should be based on the path and not the entire url. We should use `httr2::url_parse(url)$path`

Reply:
Also, any reason we can't use `gsub()` here?